integreat_cms/static/src/js/contact_box.ts

```diff
@@ -0,0 +1,106 @@
+import { getCsrfToken } from "./utils/csrf-token";
+import { createIconsAt } from "./utils/create-icons";
+
+const hideContactFormWidget = () => {
+    const widget = document.getElementById("contact-form-widget") as HTMLElement;
+    if (widget) {
+        widget.textContent = "";
+        document.getElementById("show-contact-form-button").classList.remove("hidden");
+    }
+};
+
+const addNewContactToList = (label: string, url: string) => {
+    const relatedContactList = document.getElementById("related-contact-list");
+    const newContactRow = document.createElement("a");
+
+    newContactRow.href = url;
+    newContactRow.innerHTML = `<i icon-name="pencil" class="mr-2"></i> ${label}`;
+    newContactRow.classList.add("block", "pt-2", "hover:underline");
+
+    relatedContactList.appendChild(newContactRow);
+    createIconsAt(relatedContactList);
+};
+
+const showMessage = (data: any) => {
+    const timeoutDuration = 10000;
+    if (data.success) {
+        hideContactFormWidget();
+        addNewContactToList(data.contact_label, data.edit_url);
+        const successMessageField = document.getElementById("contact-ajax-success-message");
+        successMessageField.classList.remove("hidden");
+        setTimeout(() => {
+            successMessageField.classList.add("hidden");
+        }, timeoutDuration);
+    } else {
+        const errorMessageField = document.getElementById("contact-ajax-error-message");
+        errorMessageField.classList.remove("hidden");
+        setTimeout(() => {
+            errorMessageField.classList.add("hidden");
+        }, timeoutDuration);
+    }
+};
```

Comment:
This is cool, what do you think about extracting this into its own PR and applying it globally? 😃